File: linklistimplement.py
class node:
    def __init__(self, data):
        self.data = data
        self.next = None

class stack:
    def __init__(self):
        self.top = None
    
    def push(self,x):
        newnode = node(x)
        newnode.next = self.top
        self.top = newnode

    def pop(self):
        if(self.top == None):
            return
        self.top = self.top.next
    
    def printstack(self):
        temp = self.top
        while(temp != None):
            print(temp.data, end=' ')
            temp = temp.next
        print()

s = stack()
s.push(8) # 8
s.push(3) # 3 8
s.push(5) # 5 3 8
s.push(2) # 2 5 3 8
s.printstack()
s.pop()   # 2
s.printstack()

File: linklistimplement.cpp
#include<iostream>
using namespace std;
class node
{
	public:
		int data;
		node* next;

		node(int node_data){
			this -> data = node_data;
			this -> next = nullptr;
	}
};

class queue
{
	public:
	node* front;
	node* rear;

	queue(){
		this ->front = nullptr;
		this -> rear = nullptr;
	}

	void Enqueue(int node_data){
		node* newnode = new node(node_data);
		if(this-> front == nullptr && this->rear == nullptr){
			front = rear = newnode;
		}
		this->rear->next = newnode;
		this->rear = newnode;
	}

	void Dequeue(){
		node* temp = front;
		if(front == nullptr){
			return;
		}
		if(front == rear){
			front = rear = nullptr;
		}
		else
		{
			front = front->next;
		}
		delete temp;
		
	}

	void show(){
		node* temp = front;
		while( temp != nullptr){
			cout<< temp->data << " ";
			temp = temp->next;
		}
		cout<<"\n";
	}

};

int main(){
	queue q;
	q.Enqueue(7);//7
	q.Enqueue(6);//7 6
	q.Enqueue(4);//7 6 4
	q.Enqueue(8);// 7 6 4 8
	q.show();
	q.Dequeue();//6 4 8
	q.show();
}
